File: backend/src/models/Alert.js
const mongoose = require('mongoose');

const alertSchema = new mongoose.Schema({
  user: {
    type: mongoose.Schema.Types.ObjectId,
    required: true,
    ref: 'User',
  },
  symbol: {
    type: String,
    required: true,
  },
  targetPrice: {
    type: Number,
    required: true,
  },
  direction: {
    type: String,
    enum: ['above', 'below'],
    required: true,
  },
  isTriggered: {
    type: Boolean,
    default: false,
  },
}, {
  timestamps: true,
});

module.exports = mongoose.model('Alert', alertSchema);

File: backend2/src/main/java/com/coinvista/backend/model/Alert.java
package com.coinvista.backend.model;

import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Data
@Document(collection = "alerts")
public class Alert {

    @Id
    private String id;

    private String userId;

    private String coinId;

    private String symbol;

    private String name;

    private Double targetPrice;

    private String direction; // "above" or "below"

    private boolean isTriggered = false;

    private Double lastTriggeredPrice;

    private Instant triggeredAt;

    private Instant notificationSentAt;

    @CreatedDate
    private Instant createdAt = Instant.now();
}
